### PR TITLE
v1alpha1/podnetworkconnectivitychecks: add tls info

### PR DESCRIPTION
**Pull request by sanchezl**
In order to not cause log spam when checking endpoints that expect a tls handshake, provide tls information for the check-endpoint tool to use when checking endpoints.

Note that the resource is still not required to actually validate the tls connection, this is simply needed to help the tool be a better network citizen.

### CODE REVIEW

File: operatorcontrolplane/v1alpha1/types_conditioncheck.go [hunk 1]
```diff
@@ -38,6 +39,26 @@ type PodNetworkConnectivityCheckSpec struct {
 	// +kubebuilder:validation:Pattern=`^\S+:\d*$`
 	// +required
 	TargetEndpoint string `json:"targetEndpoint"`
+
+	// TLS indicates if a tls handshake should be performed. Endpoints expecting
+	// a TLS handshake might log tls errors if a tls handshake is not attempted
+	// before closing the TCP connection. Defaults to true.
+	// +kubebuilder:default:true
+	// +optional
+	TLS bool `json:"tls,omitempty"`
```
Comment (deads2k) [inline operatorcontrolplane/v1alpha1/types_conditioncheck.go:48]: why not something like 

```go
type Spec struct{
  // optional, if specified, must contain tls.crt and tls.key.
  // if provided, the values will be used to make the tcp connection. This avoids log spam and allows additional status reports about credential viability.
  ClientCert SecretRef
}
```

**Comment by deads2k** [comment]
Can you show me roughly what this code would look like?

File: operatorcontrolplane/v1alpha1/types_conditioncheck.go (hunk 1 above)
Comment (deads2k) [reply operatorcontrolplane/v1alpha1/types_conditioncheck.go:48]: also, if the secret is missing, attempt the connection anyway.  indicate that in the godoc too.  this is good, but not fatal in cases where it is missing.

Comment (sanchezl) [reply operatorcontrolplane/v1alpha1/types_conditioncheck.go]: Done.

**Comment by sanchezl** [comment]
> Can you show me roughly what this code would look like?

@deads2k see https://github.com/openshift/cluster-kube-apiserver-operator/pull/901

**Pull request merged**